storage/src/main/java/zipkin2/storage/kafka/KafkaSpanStore.java

```diff
@@ -221,85 +266,51 @@
       this.traceId = traceId;
     }
 
-    @Override
-    List<Span> query() {
-        try {
-          final List<Span> spans = traceStore.get(traceId);
-          if (spans == null) return new ArrayList<>();
-          return spans;
-        } catch (Exception e) {
-          LOG.error("Error getting trace with ID {}", traceId, e);
-          return null;
-        }
+    @Override public List<Span> query() {
+      try {
+        final List<Span> spans = traceStore.get(traceId);
+        if (spans == null) return new ArrayList<>();
+        return spans;
+      } catch (Exception e) {
+        LOG.error("Error getting trace with ID {}", traceId, e);
+        return null;
+      }
     }
 
-    @Override
-    public Call<List<Span>> clone() {
+    @Override public Call<List<Span>> clone() {
       return new GetTraceCall(traceStore, traceId);
     }
   }
 
   static class GetDependenciesCall extends KafkaStreamsStoreCall<List<DependencyLink>> {
     final long endTs, loopback;
-    final ReadOnlyKeyValueStore<Long, DependencyLink> dependenciesStore;
+    final ReadOnlyWindowStore<Long, DependencyLink> dependenciesStore;
 
     GetDependenciesCall(long endTs, long loopback,
-        ReadOnlyKeyValueStore<Long, DependencyLink> dependenciesStore) {
+        ReadOnlyWindowStore<Long, DependencyLink> dependenciesStore) {
       this.endTs = endTs;
       this.loopback = loopback;
       this.dependenciesStore = dependenciesStore;
     }
 
-    @Override
-    List<DependencyLink> query() {
-        try {
-          Map<String, DependencyLink> dependencyLinks = new HashMap<>();
-          long from = endTs - loopback;
-          dependenciesStore.range(from, endTs)
-              .forEachRemaining(dependencyLink -> {
-                String pair = String.format("%s-%s", dependencyLink.value.parent(),
-                    dependencyLink.value.child());
-                dependencyLinks.put(pair, dependencyLink.value);
-              });
-
-          LOG.info("Dependencies found from={}-to={}: {}", from, endTs, dependencyLinks.size());
-
-          return new ArrayList<>(dependencyLinks.values());
-        } catch (Exception e) {
-          LOG.error("Error looking up for dependencies", e);
-          return new ArrayList<>();
-        }
-    }
-
-    @Override
-    public Call<List<DependencyLink>> clone() {
-      return new GetDependenciesCall(endTs, loopback, dependenciesStore);
-    }
-  }
-
-  abstract static class KafkaStreamsStoreCall<T> extends Call.Base<T> {
+    @Override public List<DependencyLink> query() {
+      try {
+        long from = endTs - loopback;
+        List<DependencyLink> dependencyLinks = new ArrayList<>();
+        dependenciesStore.fetchAll(Instant.ofEpochMilli(from), Instant.ofEpochMilli(endTs))
+            .forEachRemaining(keyValue -> dependencyLinks.add(keyValue.value));
 
-    KafkaStreamsStoreCall() {
-    }
+        LOG.info("Dependencies found from={}-to={}: {}", from, endTs, dependencyLinks.size());
```

Comment:
probably want to scrub this out eventually and/or move it up an abstraction as the inputs and outputs are not kafka specific... unless you want to know the difference between the raw links and the merged ones, then I'd expect a different line below counting that. It is an interesting use case for a metrics gauge.. hadn't really thought much about it.

Reply:
There are windows of 1 minute each counting links per pair--this is what `fetchAll` is returning--then merge is returning sum of counters; not sure how comparing both metrics would be analyzed.
I opt to reduce level to debug and count merged values.